pageserver/src/tenant/timeline.rs

```diff
@@ -3133,14 +3173,13 @@ impl Timeline {
 
     async fn find_layer(&self, layer_name: &LayerName) -> Option<Layer> {
         let guard = self.layers.read().await;
-        for historic_layer in guard.layer_map().iter_historic_layers() {
-            let historic_layer_name = historic_layer.layer_name();
-            if layer_name == &historic_layer_name {
-                return Some(guard.get_from_desc(&historic_layer));
-            }
-        }
-
-        None
+        let layer = guard
+            .layer_map()
+            .ok()?
```

Comment:
The original concern is that we must avoid incorrectly concluding a layer doesn't exist during shutdown.  So although this function is not actually used in page reconstruction afaik, isn't it generally unsafe to conflate "layer not found" and "layer map not available"?

Reply:
I think this API exists only to fulfill manual layer eviction. But you are right, it might screw up a benchmark, I will re-check this.

Reply:
[521fbd2](https://github.com/neondatabase/neon/pull/8229/commits/521fbd2674f29598d2b15080c532560156eae33c)